### PR TITLE
Recommenders 1.1.1

### PR DESCRIPTION
**Pull request by Arishamays1**
## ☆ Recommenders 1.1.1 Update ☆
[Jira Ticket](https://anaconda.atlassian.net/browse/PKG-2406)
[Upstream](https://github.com/microsoft/recommenders)
[Dependencies ](https://github.com/microsoft/recommenders/blob/main/setup.py)
# Changes
- Built package from scratch, incorporated necessary `dependencies`, `pinnings`, and `sha256`
- Due to `numba` availability, `s390x` has been skipped.
- Skipped  for python versions <3.8, and >=3.10.
- Adjusted `numpy`  pinning for compatibility with `numba`

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  skip: True  # [py<38 or py>=310 or s390x]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+
+requirements:
+  host:
+    - pip
+    - python
+    - setuptools
+    - wheel
+    - numpy 1.24.3
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
This package does not make use of numpy C-API.
Numpy doesn't need to be included in the host section.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    - pyyaml >=5.4.1,<6
+    - requests >=2.0.0,<3
+    - retrying >=1.3.3
+    - scikit-learn >=0.22.1,<1.0.3
+    - scikit-surprise >=1.0.6
+    - scipy >=1.0.0,<2
+    - seaborn >=0.8.1,<1
+    - tqdm >=4.31.1,<5
+    - transformers >=2.5.0,<5
+    - {{ pin_compatible('numpy') }}
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - numpy>=1.19
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    - wheel
+    - numpy 1.24.3
+  run:
+    - bottleneck >=1.2.1,<2
+    - category_encoders >=1.3.0,<2
+    - cornac >=1.1.2,<2
+    - jinja2 >=2,<3.1
+    - lightfm >=1.15,<2
+    - lightgbm >=2.2.1
+    - matplotlib >=2.2.2,<4
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - matplotlib-base >=2.2.2,<4
```
To not pull in Qt unnecessarily

File: recipe/meta.yaml (hunk 1 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: Thanks for the suggestions, Charles!

**Review by cbouss — APPROVED** [review]

File: recipe/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+about:
+  home: https://github.com/microsoft/recommenders
+  license: MIT
+  license_family: MIT
+  license_url: https://github.com/microsoft/recommenders/blob/main/LICENSE
+  description: |
+    This package contains functions to simplify common tasks used when
+    developing and evaluating recommender systems.
+  summary: Microsoft Recommenders - Python utilities for building recommender systems
+  doc_url: https://pypi.org/project/recommenders/
```
Comment (boldorider4) [inline recipe/meta.yaml]: I think this should be the GH repo if a better suited page doesn't exist

File: recipe/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  number: 0
+  skip: True  # [py<38 or py>=310 or s390x]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+
+requirements:
+  host:
+    - pip
+    - python
+    - setuptools
+    - wheel
```
Comment (ryanskeith) [inline recipe/meta.yaml]: Seems like it wants numpy during setup:
https://github.com/microsoft/recommenders/blob/29c64c9e8e19073da70f89f8308f6e8a2bd9fca4/setup.py#L140

ANd 

https://github.com/microsoft/recommenders/blob/29c64c9e8e19073da70f89f8308f6e8a2bd9fca4/pyproject.toml#L5

File: recipe/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+build:
+  number: 0
+  skip: True  # [py<38 or py>=310 or s390x]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+
+requirements:
+  host:
+    - pip
+    - python
+    - setuptools
```
Comment (ryanskeith) [inline recipe/meta.yaml]:
```suggestion
    - setuptools >=52
```

File: recipe/meta.yaml (hunk 5 above)
Comment (ryanskeith) [inline recipe/meta.yaml]:
```suggestion
    - wheel >=0.36
```

File: recipe/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    - requests >=2.0.0,<3
+    - retrying >=1.3.3
+    - scikit-learn >=0.22.1,<1.0.3
+    - scikit-surprise >=1.0.6
+    - scipy >=1.0.0,<2
+    - seaborn >=0.8.1,<1
+    - tqdm >=4.31.1,<5
+    - transformers >=2.5.0,<5
+    - numpy >=1.19
+
```
Comment (ryanskeith) [inline recipe/meta.yaml:48]: It always help to put these in the order you see them in setup.py or what not for easy comparison.

File: recipe/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/microsoft/recommenders/releases/download/{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 8db3785543b2ace6ddc68994e71c8920efa781432708d8d4cc7aff720f11b55f
+
+build:
+  number: 0
+  skip: True  # [py<38 or py>=310 or s390x]
```
Comment (ryanskeith) [inline recipe/meta.yaml:14]: Seems like recommenders supports more than just python 3.9 but perhaps you are just simplifying.

**Review by ryanskeith — COMMENTED** [review]
Looks good to me overall with the few pinnings in the host section that we should add.

File: recipe/meta.yaml (hunk 4 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: for the dev_url? or doc_url?

File: recipe/meta.yaml (hunk 8 above)
Comment (Arishamays1) [reply recipe/meta.yaml:14]: Based on the recommenders tag for 1.1.1, for the [setup.py](https://github.com/microsoft/recommenders/blob/b704c420ee20b67a9d756ddbfdf5c9afd04b576b/setup.py#L130), it seems that recommenders doesn't have python 3.10 compatibility at the moment.

File: recipe/meta.yaml (hunk 7 above)
Comment (Arishamays1) [reply recipe/meta.yaml:48]: got it :)

File: recipe/meta.yaml (hunk 5 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: thanks

File: recipe/meta.yaml (hunk 6 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: Thanks Ryan!

File: recipe/meta.yaml (hunk 5 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: Added!

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: `numpy >=1.19` with whitespace

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: I'm not 100% sure but the maintainers say that they require numpy at the build time https://github.com/microsoft/recommenders/commit/23ae6e0881f8d5ab1d8fd8b9b7a6f278930c28d3.
@cbouss Is this not correct and they do not depend on numpy C-API?

File: recipe/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+about:
+  home: https://github.com/microsoft/recommenders
+  license: MIT
+  license_family: MIT
+  license_url: https://github.com/microsoft/recommenders/blob/main/LICENSE
+  description: |
+    This package contains functions to simplify common tasks used when
+    developing and evaluating recommender systems.
+  summary: Microsoft Recommenders - Python utilities for building recommender systems
+  doc_url: https://pypi.org/project/recommenders/
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
  doc_url: https://microsoft-recommenders.readthedocs.io/
```

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: From what I see, recommender is only python code, and we don't need a C compiler to build it.
I stand by my first assessment.

**Review by skupr-anaconda — APPROVED** [review]
LGTM but doc_url

**Pull request merged**